omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -179,17 +175,17 @@ public void shouldEditAnAutoGenerationOptionByAnyNumberOFArguments() throws Exce
 		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
 		req.setContent(json.getBytes());
 		handle(req);
-		AutoGenerationOption updatedAutogenerationOption = identifierSourceService.getAutoGenerationOption(getId());
+		AutoGenerationOption updatedAutogenerationOption = identifierSourceService.getAutoGenerationOptionByUuid(AUTO_GENERATION_OPTION_UUID);
 		assertFalse(updatedAutogenerationOption.isAutomaticGenerationEnabled());
 		assertFalse(updatedAutogenerationOption.isManualEntryEnabled());
 	}
 	
 	@Test
 	public void shouldPurgeAnAutoGenerationOption() throws Exception {
-		assertNotNull(identifierSourceService.getAutoGenerationOption(getId()));
```

Comment:
Is there any reason for modifying these tests? They are getting by internal id, which should also be tested differently from getting by uuid

Reply:
Yes, the whole story starts with this hack: https://github.com/openmrs/openmrs-module-idgen/blob/master/api/src/main/java/org/openmrs/module/idgen/AutoGenerationOption.java#L170-L172

We need to put things write, the normal identifier used by REST Resources is the `uuid` and not the internal DB `id`.

Because the above hack existed, tests for this Resource had to be broken and a true `uuid` pattern hard to be followed.